server/src/main/java/com/adobe/testing/s3mock/domain/FileStore.java

```diff
@@ -502,11 +503,22 @@ public S3Object getS3Object(final String bucketName, final String objectName) {
 
     final List<S3Object> resultObjects = new ArrayList<>();
     final Stream<Path> directoryHierarchy = Files.walk(theBucket.getPath());
+    
+    // Determine whether the prefix ends with a path separator by looking at
+    // what adding some definitely non-separator stuff does to equality.
+    boolean endsWithSeparator = !Strings.isEmpty(prefix)
```

Comment:
don't we always have `/` as path separator here? If so, we could just use endsWith on `prefix`?

Reply:
No, this is not the case, and it illustrates what I think is another deficiency of the `FileStore` I meant to  mention, but forgot about: the S3 API lets you use just about any character (or sequence thereof) as a delimiter, so you could use e.g. a colon as a hierarchy separator. This is due to the fact that by and large, S3 treats object keys as flat Strings, not a hierarchy of folders. The hierarchy is simulated during queries with arbitrary prefixes and delimiters.
However, if you actually use a delimiter other than `/`, I suspect that FileStore will break in a few places.

This weird looking code deals with the question whether the supplied prefix ends with whatever the `Path` API accepts as a path sepatator.

IMHO the whole FileStore could be refactored into a flat list of files (most tests will use a very limited number of objects I suppose) by appropriately escaping all keys. This would make things a lot easier, I guess.

Reply:
Very good point, I'll add an issue for that: #94.

Reply:
Thanks. I just verified my suspicion with a test case using colons as separator, which indeed failed.